src/devices/GoPiGo3/GoPiGo3.cs

```diff
@@ -0,0 +1,900 @@
+
+        /// <summary>
+        /// Set the motor power in percent
+        /// </summary>
+        /// <param name="port">The Motor port to use, can be MotorLeft and/or MotorRight</param>
+        /// <param name="powerPercent">The power from - 100 to 100, or -128 for float</param>
+        public void SetMotorPower(MotorPort port, int powerPercent)
+        {
+            powerPercent = powerPercent > 127 ? (byte)127 : powerPercent;
+            powerPercent = powerPercent < -128 ? -128 : powerPercent;
+            byte bPower = (byte)(powerPercent & 0xFF);
+            byte[] outArray = { SpiAddress, (byte)SpiMessageType.SetMotorPower, (byte)port, bPower };
+            var ret = SpiTransferArray(outArray);
+        }
+
+        /// <summary>
+        /// Set the motor target position in degrees
+        /// </summary>
+        /// <param name="port">The Motor port to use, can be MotorLeft and/or MotorRight</param>
+        /// <param name="positionDegree">The target position in degree</param>
+        public void SetMotorPosition(MotorPort port, int positionDegree)
+        {
+            byte[] outArray = { SpiAddress, (byte)SpiMessageType.SetMotorPosition, (byte)port, (byte)((positionDegree >> 24) & 0xFF), (byte)((positionDegree >> 16) & 0xFF), (byte)((positionDegree >> 8) & 0xFF), (byte)(positionDegree & 0xFF) };
+            var ret = SpiTransferArray(outArray);
+        }
+
+        /// <summary>
+        /// Set the motor target position KP constant
+        /// If you set KP higher, the motor will be more responsive to errors in position, at the cost of perhaps overshooting and oscillating.
+        /// KP slows down the motor as it approaches the target, and helps to prevent overshoot.
+        /// In general, if you increase KP, you should also increase KD to keep the motor from overshooting and oscillating.
+        /// </summary>
+        /// <param name="port">The Motor port to use, can be MotorLeft and/or MotorRight</param>
+        /// <param name="kp">The KP constant (default 25)</param>
+        public void SetMotorPositionKP(MotorPort port, byte kp = 25)
+        {
+            byte[] outArray = { SpiAddress, (byte)SpiMessageType.SetMotorPositionKp, (byte)port, kp };
+            var ret = SpiTransferArray(outArray);
+        }
+
+        /// <summary>
+        /// Set the motor target position KD constant
+        /// If you set KP higher, the motor will be more responsive to errors in position, at the cost of perhaps overshooting and oscillating.
+        /// KD slows down the motor as it approaches the target, and helps to prevent overshoot.
+        /// In general, if you increase kp, you should also increase KD to keep the motor from overshooting and oscillating.
+        /// </summary>
+        /// <param name="port">The Motor port to use, can be MotorLeft and/or MotorRight</param>
+        /// <param name="kd">The KD constant (default 70)</param>
+        public void SetMotorPositionKD(MotorPort port, byte kd = 70)
```

Comment:
interesting there is no integral factor exposed in the motor

Reply:
Yes, I realize there is a mismatch in the description. Most likely a bad copy/paste from me as both description as the smae